### PR TITLE
add sumologic_installed_collector resource

### PR DESCRIPTION
**Pull request by vsinghal13**
This PR adds the `sumologic_installed_collector` resource to be managed by Terraform. This can only be used to read, update and delete an already existing collector which can be imported in terraform.

- [X] Tested locally by importing an existing installed collector.
- [X] Able to assign new fields
- [X] Able to assign budget via `_budget` field
- [X] Able to delete the collector

Import statement: `terraform import sumologic_installed_collector.installed_collector 1xxxxxx`

Sample resource block:
```
resource "sumologic_installed_collector" "installed_collector" {
  name        = "vsinghal-mac-tf-new"
  category = "macos/vsinghal"
  ephemeral = true
  fields =  {
      _budget =  "Dev_20GB"
      key = "value"

    }
}
```

Sample tfstate:
```
{
  "version": 4,
  "terraform_version": "1.1.7",
  "serial": 31,
  "lineage": "ed6e03d2-f9ed-28a7-1cac-34bbdcb9160d",
  "outputs": {},
  "resources": [
    {
      "mode": "managed",
      "type": "sumologic_installed_collector",
      "name": "installed_collector",
      "provider": "provider[\"sumologic.com/dev/sumologic\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "alive": true,
            "category": "macos/vsinghal",
            "collector_version": "19.392-5",
            "cutoff_timestamp": 0,
            "description": "",
            "ephemeral": true,
            "fields": {
              "_budget": "Dev_20GB",
              "key": "value"
            },
            "host_name": "",
            "id": "1xxxxx",
            "last_seen_alive": 1651785111049,
            "name": "vsinghal-mac-tf-new",
            "source_sync_mode": "UI",
            "target_cpu": null,
            "timezone": "Etc/UTC"
          },
          "sensitive_attributes": [],
          "private": "bnVsbA=="
        }
      ]
    }
  ]
}
```

### CODE REVIEW

**Comment by seansain** [comment]
Let's add some docs? LGTM otherwise.

File: sumologic/resource_sumologic_installed_collector.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	id, _ := strconv.Atoi(d.Id())
+
+	return Collector{
+		ID:            id,
+		CollectorType: "Installable",
+		Name:          d.Get("name").(string),
+		Description:   d.Get("description").(string),
+		Category:      d.Get("category").(string),
+		TimeZone:      d.Get("timezone").(string),
+		Fields:        d.Get("fields").(map[string]interface{}),
```
Comment (maimaisie) [inline sumologic/resource_sumologic_installed_collector.go:71]: what about `hostName`, `logIngestStatus`, `ephemeral`, `tartgetCpu` and `sourceSyncMode`? (might not be exhaustive)

Comment (vsinghal13) [reply sumologic/resource_sumologic_installed_collector.go:71]: will add modifiable fields mentioned here: https://help.sumologic.com/APIs/Collector-Management-API/Collector-API-Methods-and-Examples#response-fields

**Comment by vsinghal13** [comment]
Acceptance tests cannot be added for this resource since tf doesn't support importing the resource as the first step in the acceptance test `TestStep`  and we will be doing manual QE verification.

**Comment by vsinghal13** [comment]
@AvadheshKaria @maimaisie @hchoudharysumologic ping for review.

**Review by AvadheshKaria — APPROVED** [review]

**Comment by AvadheshKaria** [comment]
Can we add test for the same, similar to the tests we have in *_test.go files for other features.

File: website/docs/r/installed_collector.html.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+**NOTE**: This resource can only be imported and managed by terraform. Creation of this resource via terrform is not supported.
+
+## Example Usage
+```hcl
+resource "sumologic_installed_collector" "installed_collector" {
+  name        = "vsinghal-mac"
+  category = "macos/vsinghal"
+  ephemeral = true
+  fields =  {
+      _budget =  "Dev_20GB"
```
Comment (maimaisie) [inline website/docs/r/installed_collector.html.markdown:19]: `_budget` is v1 budget which is not recommended for customers anymore, use a different example?

**Review by maimaisie — APPROVED** [review]
a few minor comments about the example used, else lgtm

**Comment by frek818** [comment]
What's holding this merge back? Is that anything I can do to help?

**Comment by vsinghal13** [comment]
@frek818 this is pending QE verification from our end.

**Pull request merged**